### PR TITLE
Consensus feature

### PR DESCRIPTION
**Pull request by bokobza**
Moved all the Consensus code to its own feature.
Added Consensus logger.
Moved blockStoreManager flushing out of the full node and into the feature.

### CODE REVIEW

**Comment by bokobza** [comment]
Unit tests are passing and the node runs nicely :-)

File: Stratis.Bitcoin/Builder/FullNodeBuilder.cs [hunk 1]
```diff
@@ -70,9 +70,7 @@ public FullNodeBuilder(NodeSettings nodeSettings) : base()
 			{
 				service.AddSingleton(this.NodeSettings);
 				service.AddSingleton(this.Network);
-			});
-
-			this.AddRequired();
```
Comment (dangershony) [inline Stratis.Bitcoin/Builder/FullNodeBuilder.cs]: UseBase() needs to go instead

File: Stratis.Bitcoin/Consensus/ConsensusFeature.cs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        public static IFullNodeBuilder UseConsensus(this IFullNodeBuilder fullNodeBuilder)
+        {
+            
+            fullNodeBuilder.ConfigureFeature(features =>
+            {
+                features
+                .AddFeature<ConsensusFeature>()
+                .FeatureServices(services =>
+                {
+                    var dataFolder = new DataFolder(fullNodeBuilder.NodeSettings);
```
Comment (dangershony) [inline Stratis.Bitcoin/Consensus/ConsensusFeature.cs]: This should be injected but I guess that's fine for now @Neurosploit is fixing this in his next commit.

File: Stratis.Bitcoin/Logging/Logs.cs [hunk 3]
```diff
@@ -58,6 +58,12 @@ public static ILogger EstimateFee
 		{
 			get; set;
 		}
-		public const int ColumnLength = 16;
+
+        public static ILogger Consensus
```
Comment (dangershony) [inline Stratis.Bitcoin/Logging/Logs.cs]: Indent

**Pull request merged**